sky/utils/controller_utils.py

```diff
@@ -151,9 +152,9 @@ def _get_cloud_dependencies_installation_commands(
         # dependencies for sky serve controller.
         commands.append('pip list | grep oci > /dev/null 2>&1 || '
                         'pip install oci > /dev/null 2>&1')
-    else:
-        # We do not install azure dependencies for spot controller since our
-        # subscription does not support spot instances.
+    # TODO(tian): Make dependency installation command a method of cloud class
+    # and get all installation command for enabled clouds.
+    if clouds.Azure() in global_user_state.get_enabled_clouds():
```

Comment:
won't this always be `False? clouds.Azure() is a different object then the one in `get_enabled_clouds()`. Check out the function: https://github.com/skypilot-org/skypilot/blob/1bb59fd25b7b2de9ccb65c52c42a356145aaddb7/sky/optimizer.py#L1088-L1091

Reply:
Make sense! Thanks. Fixed!